aip/general/0162.md

```diff
@@ -27,199 +27,174 @@ to the version of an API as a whole.
-it would not make sense (or lead to dangerous mistakes).
+Depending on the resource, different APIs may have different strategies for
+when to create a new revision, such as:
 
-### Getting a revision
+- Create a new revision any time that there is a change
+- Create a new revision when something important happens
+- Create a new revision when the user specifically asks
 
-APIs implementing resource revisions **should** accept a resource name with a
-revision ID in the standard `Get` method ([AIP-131][]):
+APIs **may** use any of these strategies. APIs that want to create a revision on
+user request should handle this with a custom method on the resource:
 
 ```proto
-message GetBookRequest {
-  // The name of the book.
-  //   Example: publishers/123/books/les-miserables
-  //
-  // In order to retrieve a previous revision of the book, also provide
-  // the revision ID.
-  //   Example: publishers/123/books/les-miserables@c7cfa2a8
+rpc CommitBookRevision(CommitBookRevisionRequest) returns (BookRevision) {
+  option (google.api.http) = {
+    post: "/v1/{name=publishers/*/books/*}:commitRevision"
+    body: "*"
+  };
+}
+
+message CommitBookRequest {
   string name = 1 [
     (google.api.field_behavior) = REQUIRED,
     (google.api.resource_reference) = {
       type: "library.googleapis.com/Book"
     }];
+  string tag = 2 [(google.api.field_behavior) = OPTIONAL];
 }
 ```
 
-If the user passes a revision ID that does not exist, the API **must** fail
-with a `NOT_FOUND` error.
+- The method **must** use the `POST` HTTP verb.
+- The method **should** return a long running operation or resource revision.
+- The request message **must** include the `name` field.
+- The resource ID of the resource revision **should** be a string and contain a
+  short, automatically-generated random string. A good rule of thumb is the last
+  eight characters of a [UUID version 4 (random)](UUID4).
+- `tag` field is optional. If provided, revision created will have the tag set.
+
+### Getting a revision
 
-APIs **must** return a `name` value corresponding to what the user sent. If the
-user sent a name with no revision ID, the returned `name` string **must not**
-include the revision ID either. Similarly, if the user sent a name with a
-revision ID, the returned `name` string **must** explicitly include it.
+Services `must` implement the standard `Get` method ([AIP-131][])
 
 ### Tagging revisions
 
+There are two types of tags: server and user specified.
+
+#### Server specified
+
+Services **may** reserve specific tags to be server-specified (e.g. `current`,
+`latest`), which are read-only and managed by the service.
+
+- If a `latest` tag exists, it **must** represents the most recent created
+  revision.
+ The content of `publishers/{publisher}/books/{book}/revisions/latest` and
+  `publishers/{publisher}/books/{book}` can differ. Because the latest revision
+  may be different from the current state of the resource.
```

Comment:
Does this mean it is up to the implementation to decide what a GET on the resource (as in no child revision specified) means?

So one could allow user specific revisions (such as the custom commit API example) which would not necessarily have its changes reflected on what is returned by actual resource?

Reply:
a GET on the resource should return the resource in it's current state. That's true regardless of revisions. 

> So one could allow user specific revisions (such as the custom commit API example) which would not necessarily have its changes reflected on what is returned by actual resource?

Correct, that is the purpose of revisions: to store the state of a resource at a point in time. See "When resource revisions are useful" for an enumeration of use cases.

Did you have a different expectation from a revisions collection?

Reply:
I think what I didn't quite understand is the fact that the documentation mentions that `latest` is not necessarily the same as `current`. If so, what is the definition of `current`? Is that left to the API to define?

Regarding the CustomCommit question, I think I just need to re-read the changes again. Almost feels like one could allow for requests where the user supplies the revision value through a custom method and if that's the case why have a concept of tags? Probably missed something somewhere.